bpt_visitor.cpp

```diff
@@ -0,0 +1,52 @@
+#include "bpt_visitor.hpp"
+#include "bpt_events.hpp"
+
+namespace bpt {
+
+void visitor::visit(const modload_event& e) {
+    visit(static_cast<const event&>(e));
```

Comment:
Why do you need this functions, and why do you use `static_cast` here if it is the upcast?